interpreter/valid/valid.ml

```diff
@@ -139,7 +143,8 @@ let check_heap_type (c : context) (t : heap_type) at =
   | ExnHT | NoExnHT
   | ExternHT | NoExternHT -> ()
   | VarHT (StatX x) -> let _dt = type_ c (x @@ at) in ()
-  | VarHT (RecX _) | DefHT _ -> assert false
+  | VarHT (RecX _) -> assert false
+  | DefHT dt -> check_def_type c dt at
```

Comment:
Is'm a bit confused about this, is this case actually hit with the program you give?

Reply:
This is actually the only error that appears (my first fix was to recursively check the argument, ie something like `check_sub_type (expand_def_type dt)`, which then hit the continuation issue; I thought about the issue a bit more after fixing that second error and this `check_def_type` is what I came up with in the end).